src/unix/stream.c

```diff
@@ -1140,6 +1140,11 @@ static void uv__read(uv_stream_t* stream) {
         }
         stream->read_cb(stream, 0, &buf);
       } else {
+        if ((errno == ECONNRESET) && (stream->flags & UV_STREAM_DISCONNECT)) {
+          uv__stream_eof(stream, &buf);
+          return;
```

Comment:
Seeing how you have a return statement here, you might as well fold the if statement into the if/else statement above it (and then you don't need the return statement either because there is one below.)